src/utils/logger/index.ts

```diff
@@ -0,0 +1,119 @@
+import { Logger, transports, LoggerInstance } from 'winston';
+import * as path from 'path';
+
+import { LoggerMethods, Loggers, LogMessage } from './types';
+import levels from './levels';
+
+let loggers: Loggers = null;
+
+/**
+ * Resolve path for filename in <root>/logs
+ * @param {string} filename the name of the log file
+ */
+function resolve(filename: string) {
+  return path.resolve('..', '..', '..', 'logs', filename);
+}
+
+/**
+ * Perform check to ensure that loggers are initialized before usage
+ */
+function check() {
+  if (!loggers) {
+    throw new Error('Logger was not initialized');
+  }
+}
+
+/**
+ * Generate a logger dynamically
+ *
+ * @param {string} name Name of logger to create
+ */
+function generateLogger(name: string) {
+  return {
+    [name]: new (Logger)({
+      transports: [
+        new (transports.File)({
+          name: `${name}-file`,
+          filename: resolve(`server-${name}.log`),
+          level: name,
+          json: true,
+          maxsize: 5242880,
+          maxFiles: 5,
+          handleExceptions: handleExceptions(name),
+          colorize: false,
+        }),
+
+        new (transports.Console)({
+          handleExceptions: handleExceptions(name),
+          colorize: true,
+        }),
+      ],
+    }),
+  };
+}
+
+function handleExceptions(name: string) {
+  switch (name) {
+    case 'error':
+      return true;
+
+    default:
+      return false;
+  }
+}
+
+/** Prepare objects for the logger */
+function prepareMessage(...msg: LogMessage[]): string {
+  return msg.map(m => JSON.stringify(m)).join('\n');
+}
+
+/**
+ * Create a logger method that uses logger of 'name'
+ *
+ * @param {string} name Name of logger method to create
+ */
+function generateLoggerMethod(name: string) {
```

Comment:
Is this called everytime we want to log something? If it is, I thin ka shorter name would be nice. `genLogMethod`?

Reply:
Nah this is just to generate them all at first, the module exposes a logger object that you can use